docs/maven-tutorial.md

```diff
@@ -0,0 +1,150 @@
+## Prerequisites
+
+This tutorial assumes that you'll be working with a project that is already set
+up for use with Maven. If you're interested in using Maven with your Java
+project, we highly recommend looking at
+[their guide](https://maven.apache.org/guides/getting-started/index.html).
+
+In addition, be sure to check your application's image for Maven mirror
+capabilities. Many images that use Maven have a 'MAVEN_MIRROR_URL' environment
+variable that you can use to simplify this process. If it doesn't have this
+capability, read
+[the Nexus Maven documentation](https://books.sonatype.com/nexus-book/reference/config.html)
+to configure your build properly.
+
+## Setting up Nexus
+
+Use `new-app` to download and deploy the official Nexus Docker image.
+```bash
+oc new-app sonatype/nexus
+```
+
+After the build has finished, run:
+```bash
+oc get svc
+```
+
+The output should resemble:
+```bash
+NAME      CLUSTER-IP       EXTERNAL-IP   PORT(S)    AGE
+nexus     172.30.170.249   <none>        8081/TCP   6s
+```
+
+Confirm that Nexus is running by navigating your browser to the new Cluster-IP
+at port 8081. The server may take a bit of time to start, be patient!
+
+> NOTE: Nexus comes pre-configured for the Central Repository, but you may need
+> others for your application. At this point, add any special repositories to
+> Nexus. Documentation on how to add resositories can be found
+> [here](https://books.sonatype.com/nexus-book/reference/config-sect-new-repo.html).
+
+#### Using Probes to Check for Success
+
+This is a good time to set up readiness and livenes probes. These will
+periodically check to see that Nexus is running properly. More information on
+probes can be found
+[here](https://docs.openshift.org/latest/dev_guide/application_health.html).
+
+```bash
+oc set probe dc/nexus \
+	--liveness \
+	--failure-threshold 3 \
+	--initial-delay-second 30 \
+	-- echo ok
+oc set probe dc/nexus \
+	--readiness \
+	--failure-threshold 3 \
+	--initial-delay-seconds 30 \
+	--get-url=http://nexus:8081/content/groups/public
+```
+
+#### Adding Persistence to Nexus
+
+> NOTE: If you don't want persistent storage, continue to
+> [Connecting to Nexus](#connecting-to-nexus). However, your cached
+> dependencies will be lost if the pod is restarted for any reason.
+
+Now we'll create a Persistent Volume Claim for Nexus, so that our dependencies
+will not be lost when the Pod running the server terminates. If you do not have
+admin access on your cluster, ask your system administrator to create a
+Read/Write Persistent Volume for you.  Otherwise, read
+[Managing Volumes in OpenShift](https://docs.openshift.org/latest/dev_guide/volumes.html)
+for instructions on creating a persistent volume.
+
+Add a Persistent Volume Claim to the Nexus Deployment Config.
+```bash
+oc volumes dc/nexus --remove --confirm
+oc volumes dc/nexus --add \
+	--name 'nexus-pvc' \
+	--type 'pvc' \
+	--mount-path '/sonatype-work/' \
+	--claim-name 'nexus-pv' \
+	--claim-size '1G'
```

Comment:
hm, i guess i don't know, i was going off what we actually put into the json resource definition.  if the help for the cli shows G, use that and ignore me.